Sources/SnapshotTesting/Snapshotting/NSImage.swift

```diff
@@ -77,14 +79,19 @@ private func compare(_ old: NSImage, _ new: NSImage, precision: Float) -> Bool {
   let newRep = NSBitmapImageRep(cgImage: newerCgImage)
   var differentPixelCount = 0
   let pixelCount = oldRep.pixelsWide * oldRep.pixelsHigh
-  let threshold = (1 - precision) * Float(pixelCount)
+  let threshold = Int((1 - precision) * Float(pixelCount))
   let p1: UnsafeMutablePointer<UInt8> = oldRep.bitmapData!
   let p2: UnsafeMutablePointer<UInt8> = newRep.bitmapData!
-  for offset in 0 ..< pixelCount * 4 {
-    if p1[offset] != p2[offset] {
-        differentPixelCount += 1
+
+  var offset = 0
+  while offset < pixelCount * 4 {
+    if p1[offset].diff(between: p2[offset]) > subpixelThreshold {
+      differentPixelCount += 1
+      if differentPixelCount > threshold {
+        return false
+      }
     }
-    if Float(differentPixelCount) > threshold { return false }
+    offset += 1
```

Comment:
Is there a reason this was changed from a for-loop to a while-loop? My interpretation is that in both implementations the loop should either increment `offset` or `return false`.

Using a while loop and manually incrementing the `offset` makes this less maintainable since it's easy to introduce changes in the loop body that skip incrementing `offset` and would lead to an infinite loop.

Reply:
Nevermind, just saw your PR description with the explanation for when compiled without optimization.